src/core/services-impl/src/main/java/it/geosolutions/geostore/util/SearchConverter.java

```diff
@@ -242,6 +241,36 @@ public void visit(CategoryFilter filter) throws InternalErrorServiceEx {
         trgFilter = f;
     }
 
+    /** This is a leaf filter. */
+    @Override
+    public void visit(GroupFilter filter) {
+        GroupFilter.checkOperator(filter.getOperator());
+
+        Integer op = ops_rest_trg.get(filter.getOperator());
+
+        if (op == null) {
+            throw new IllegalStateException("Unknown op " + filter.getOperator());
+        }
+
+        Filter f = new Filter();
+        f.setOperator(op);
+        f.setProperty("security.group.groupName");
+
+        List<String> names = filter.getNames();
+
+        if (SearchOperator.IN != filter.getOperator() && names.size() != 1) {
+            throw new IllegalStateException("Erroneous search op " + filter.getOperator());
+        }
```

Comment:
For IN, you require names.size() != 1. This is good, but you might also want to handle “empty list” vs. “null.”

You do have tests for GroupFilter in RESTExtJsServiceImplTest. Make sure you also test edge cases: empty names, null operator, etc.

Reply:
This condition is actually checking filter values for operators that are not `IN` (e.g. for `EQUAL_TO` assures that we have just one value to avoid ambiguities).
Added some edge cases tests anyway.